service/src/test/org/apache/hive/service/cli/thrift/ThriftCLITestUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hive.service.cli.thrift;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hive.service.server.HiveServer2;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+import org.junit.BeforeClass;
+
+public abstract class ThriftCLITestUtils {
```

Comment:
i am not sure if we should extend from it or just use it as Utils class

Reply:
I have made the setup base class as static. Keeping this class as abstract as ThriftCLIServiceTest class followed by other test classes inherit the variables declared here. And the code was getting pretty cluttered on doing so.

I have also removed the explicit calls of member variables.

Reply:
all vars are static, why do you need to inherit here?